servers/src/main/java/tachyon/master/block/BlockMaster.java

```diff
@@ -324,20 +334,18 @@ public void removeBlocks(List<Long> blockIds) {
   }
 
   /**
-   * @return a new block container id. Called by internal masters
+   * @return a new block container id
    */
   @Override
   public long getNewContainerId() {
-    synchronized (mBlockContainerIdGenerator) {
-      long containerId = mBlockContainerIdGenerator.getNewContainerId();
-      writeJournalEntry(mBlockContainerIdGenerator.toJournalEntry());
-      flushJournal();
-      return containerId;
-    }
+    long containerId = mBlockContainerIdGenerator.getNewContainerId();
```

Comment:
I think we still need `synchronized (mBlockContainerIdGenerator)`. The reason is that, without this synchronization, if two `getNewContainerId` invokes are interleaved, we will end up with wrong journals